src/main/java/Parser.java

```diff
@@ -0,0 +1,91 @@
+package main.java;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeParseException;
+
+import main.java.command.AddDeadlineCommand;
+import main.java.command.AddEventCommand;
+import main.java.command.AddTodoCommand;
+import main.java.command.Command;
+import main.java.command.DeleteCommand;
+import main.java.command.DoneCommand;
+import main.java.command.ExitCommand;
+import main.java.command.FindCommand;
+import main.java.command.ListCommand;
+import main.java.command.WrongCommand;
+import main.java.exceptions.IllegalInputFormatException;
+
+/**
+ * A Parser class that is used to parse user input as Commands
+ */
+public class Parser {
+
+    /**
+     * Creates a parser instance
+     */
+    public Parser() {
+
+    }
+
+    /**
+     * Parse a particular command string and create a command to be returned
+     * @param commandStr user input string command
+     * @return parsed command
+     * @throws Exception parsing exceptions including wrong input, index out of bounds due to wrong input etc.
+     */
+    public Command parse(String commandStr) {
```

Comment:
Maybe you could consider avoiding long methods as the textbook says to be wary of methods that go beyond 30 LOC? This could improve the readability of the code.